infoschema/builder.go

```diff
@@ -190,7 +192,7 @@ type Builder struct {
 
 // ApplyDiff applies SchemaDiff to the new InfoSchema.
 // Return the detail updated table IDs that are produced from SchemaDiff and an error.
-func (b *Builder) ApplyDiff(m *meta.Meta, diff *model.SchemaDiff) ([]int64, error) {
+func (b *Builder) ApplyDiff(store kv.Storage, m *meta.Meta, diff *model.SchemaDiff) ([]int64, error) {
```

Comment:
We could use the [`store` member of `Builder`]([https://github.com/pingcap/tidb/blob/9c5228517ccdbdd3c59b52ceb2a568ce4d6ac7b0/infoschema/builder.go/#L187), so that you don't need to pass it for `ApplyDiff`, `applyCreateTables` and `applyExchangeTablePartition`.

Reply:
good suggestion!